src/webportal/src/app/job-submission/components/tab-form.jsx

```diff
@@ -59,28 +61,44 @@ export class TabForm extends React.Component {
     return Number(key.substring(TAB_ITEM_KEY_PREFIX.length));
   }
 
-  _renderPivotItems(items) {
-    const pivotItems = items.map((items) =>
-                         <PivotItem key={items.itemKey}
-                                    itemKey={items.itemKey}
-                                    headerText={items.headerText}
-                                    onRenderItemLink={this._onRenderItem.bind(this)}/>);
-
-    return pivotItems;
-  }
-
-  _onRenderItem(itemPros, defaultRender) {
-    if (itemPros === undefined || defaultRender === undefined) {
+  _onRenderItem(itemProps, defaultRender) {
+    if (itemProps === undefined || defaultRender === undefined) {
       return null;
     }
 
+    const {spacing, palette} = getTheme();
+    const {items} = this.props;
+    const {selectedIndex} = this.state;
+    const idx = this._getItemIndexByKey(itemProps.itemKey);
+    const item = items[idx];
+    const taskRolesObject = item.content.convertToProtocolFormat();
+    const {error} = Joi.validate(taskRolesObject, taskRolesSchema);
```

Comment:
May be we should also validate the prerequisites which associate with this taskRole here.

Reply:
I think you mean docker info. Is there anything else could be edited in task-role section?

Reply:
I think that's enough, other fields can be validated by taskRole JOI. Thanks for this change.